### DISCUSSION TITLE
Searching for a string parameter

### DISCUSSION THREAD

**Discussion by aaisha79** [BurntSushi/ripgrep#2084] [Q&A] [answered]
Hi there

If i am trying to search for a string in a binary file which is on 2 lines in the following format whats the best way to approach

				<key>abc</key>
		<string>abcdef</string>

I have tried the following however its failing "(?i)<key>abc</key>\\s+<string>abcdef</string>"

**Comment by BurntSushi** [comment, marked as answer]
Well, ripgrep is a line oriented search tool. So searching across lines requires enabling multi-line mode. To that end, the simple answer to your question is to write a regex that searches across multiple lines:

```
$ rg -U 'abc.*?\n.*?abcdef' /tmp/rg-2084.txt
1:                      <key>abc</key>
2:      <string>abcdef</string>
```

Your regex doesn't account for the fact that there are non-whitespace characters between `abc` and `abcdef`.

**Reply by aaisha79** [reply]
Thanks for that will give it a try.

**Comment by aaisha79** [comment]
Hi so i tried using the following syntax 'abc.*?\n.*?abcdef' using multi line mode but i got the following error

[ERROR][GrepEngine] Could not parse ripgrep response: 
 [the literal '"\n"' is not allowed in a regex

**Reply by BurntSushi** [reply]
Why not show me the full command? Are you actually using the `-U` flag?

**Reply by aaisha79** [reply]
Apologies so my full JSON fragment looks like the following:

"engineType": "GREP",
"match_value":'abc.*?\n.*?abcdef'
"file_pattern": "abcdf.txt",
        "invert": true,
        "matcher_options":{
          "search_multiline": true

**Reply by BurntSushi** [reply]
That's not a command. That's a JSON fragment. Look at what I posted:

```
$ rg -U 'abc.*?\n.*?abcdef' /tmp/rg-2084.txt
1:                      <key>abc</key>
2:      <string>abcdef</string>
```

`rg -U 'abc.*?\n.*?abcdef' /tmp/rg-2084.txt` is the command there. You need to share which command you're running. If you're using some sort of wrapper around ripgrep that I'm unfamiliar with and don't support, then I'm not sure how to help you.

You might consider that there is a multi-line mode for searching, and _also_ a regex multiline option. You are probably enabling the latter (which is enabled by default anyway) and not the former. When using `rg`, you enable the former with the `-U` flag. I don't know how you enable it with whatever it is you're doing.

In the future, when asking questions, it would be super helpful for you to share all of the context surrounding how you're using the tool.

**Reply by aaisha79** [reply]
Sure no worries thanks
